packages/odyssey-react/src/utils/cx.ts

```diff
@@ -0,0 +1,38 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { useMemo } from 'react';
+
+type arg = string | boolean | undefined | { [key: string]: boolean | undefined }
+
+type cx = (...args: arg[]) => string;
+
+export const cx: cx = (...args) => {
+  let classNames = ''
+  let lead = ''
+
+  for (const arg of args) {
+    if (typeof arg === 'string') { classNames += `${lead}${arg}` }
+    if (typeof arg === 'object') {
+      Object.entries(arg).forEach(
+        ([k, v]) => v && (classNames +=`${lead}${k}`)
+      )
+    }
+    if (!lead) { lead = ' ' }
+  }
+
+  return classNames;
+}
+
+export const useCx: cx = (...args) => {
```

Comment:
Not sure if this is convention in this repo, but JSDoc would be super helpful for devs with intelliSense.

Also - with utility functions like this get a lot of value when we had some documentation. This helps contributors know that we have our own utility for `classnames`, so it should not be added as a new dependency.

Reply:
good thought! It is something I'd like to revisit in the future though after we get beyond our current focus. In the interim, tests and type annotations will provide a healthy amount of documentation for those who are curious